components/centraldashboard/app/api.ts

```diff
@@ -181,33 +268,170 @@ export class Api {
+                    namespaces[namespace].owner,
+                    namespaces[namespace].contributors.join(', '),
+                  ]);
+                  res.json(tabular);
+                } catch(err) {
+                  this.surfaceProfileControllerErrors({
+                    res,
+                    msg: `Unable to fetch all workgroup data`,
+                    err,
+                  });
+                }
+              })
+        .get(
+            '/get-contributors/:namespace',
+            async (req: Request, res: Response) => {
+              const {namespace} = req.params;
+              try {
+                const users = await this.getContributors(namespace);
+                res.json(users);
+              } catch(err) {
+                this.surfaceProfileControllerErrors({
+                  res,
+                  msg: `Unable to fetch contributors for ${namespace}`,
+                  err,
+                });
+              }
+            })
         .post('/create-workgroup', async (req: Request, res: Response) => {
-          if (!req.user.hasAuth) {
-            res.status(405).json(OPERATION_NOT_SUPPORTED);
-            return;
-          }
-
-          const profile = req.body as CreateProfileRequest;
-          try {
-            // Use the request body if provided, fallback to auth headers
-            await this.profilesService.createProfile({
-              metadata: {
-                name: profile.namespace || req.user.username,
-              },
-              spec: {
-                owner: {
-                  kind: 'User',
-                  name: profile.user || req.user.email,
+              if (!req.user.hasAuth) {
+                return apiError({
+                  res,
+                  code: 405,
+                  error: ERRORS.operation_not_supported,
+                });
+              }
+    
+              const profile = req.body as CreateProfileRequest;
+              try {
+                const namespace = profile.namespace || req.user.username;
+                // Use the request body if provided, fallback to auth headers
+                await this.profilesService.createProfile({
+                  metadata: {
+                    name: namespace,
+                  },
+                  spec: {
+                    owner: {
+                      kind: 'User',
+                      name: profile.user || req.user.email,
+                    }
+                  },
+                });
+                res.json({message: `Created namespace ${namespace}`});
+              } catch (err) {
+                this.surfaceProfileControllerErrors({
+                  res,
+                  msg: 'Unexpected error creating profile',
+                  err,
+                });
+              }
+            })
+        .post(
+              '/add-contributor/:namespace',
+              async (req: Request, res: Response) => {
+                const {namespace} = req.params;
```

Comment:
It looks like the logic for add and remove contributor starts out very similarly. It would probably be good to move that to a helper method. Also, maybe it would make sense to have use the `route` method to group `get`, `post`, and `delete` methods under a common path for contributors.

Reply:
Done!